### PR TITLE
Updated documentation - Visual Studio specificities, Shield compat table, what's new update

### PR DESCRIPTION
**Pull request by IoTDan**
New documentation for a shield compatibility table and info on some "new to visual studio" specificities.

### CODE REVIEW

File: ShieldCompat.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+| [Sunfounder flat RGB LED](http://www.amazon.com/SunFounder-modules-Arduino-Mega2560-Mega328/dp/B00CBDUD60){:target="_blank"} | None | None |
+| [Sunfounder Relay Module](http://www.amazon.com/SunFounder-modules-Arduino-Mega2560-Mega328/dp/B00CBDUD60){:target="_blank"} | None | [Arudino Forum](http://forum.arduino.cc/index.php?topic=142207.0){:target="_blank"} |
+| [Tower Pro MG90S Microservo (with Adafruit PWM Driver)](http://www.valuehobby.com/mg90s-mini-servo.html){:target="_blank"} | [Adafruit Driver Library](https://learn.adafruit.com/16-channel-pwm-servo-driver/using-the-adafruit-library){:target="_blank"} | [Adafruit PWM driver tutorial](https://learn.adafruit.com/16-channel-pwm-servo-driver){:target="_blank"} |
+| [Universal 16-key Keypad](http://www.amazon.com/Universial-Switch-Keypad-Keyboard-Arduino/dp/B008A30NW4/ref=cm_cr_pr_product_top){:target="_blank"} | [Arduino Keypad Library](http://playground.arduino.cc/code/Keypad){:target="_blank"} | [Key Matricies Info](http://pcbheaven.com/wikipages/How_Key_Matrices_Works/){:target="_blank"} |
+
+<br/>
+<a class="btn btn-default" href="index.htm" role="button">Return to homepage</a>
```
Comment (ooeygui) [inline ShieldCompat.md]: ## Can you update so it uses the markdown version of this?

`[&laquo; Return to Samples](SampleApps.htm){: .btn .btn-default}`

File: VisualStudio.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+#Precompiled Headers
+While a new concept to Wiring and Sketches, precompiled headers are common in [C and C++ development](http://en.wikipedia.org/wiki/Precompiled_header){:target="_blank"}.
+They make compilation even more efficient than using #pragmas.
+
+The default Wiring template creates a few files (stdafx.h and stdafx.cpp) in your project that are used to manage precompiled headers.
+You can find more info about these files [here](http://msdn.microsoft.com/en-us/library/h552b3ca.aspx){:target="_blank"}.
+
+<br/>
+<a class="btn btn-default" href="index.htm" role="button">Return to homepage</a>
```
Comment (ooeygui) [inline VisualStudio.md]: `[&laquo; Return to Samples](SampleApps.htm){: .btn .btn-default}`

File: VisualStudio.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+While non standard in C, C++ and Wiring, it is quite common to use this #pragma over the standard #ifndef/#ifdef/#endif approach.
+Using #pragma allows the Visual Studio compiler to efficiently open/read/close the file if #pragma once is encountered over the standard approach.
+This allows compile time of your program/sketch to be improved.
+
+#Precompiled Headers
+While a new concept to Wiring and Sketches, precompiled headers are common in [C and C++ development](http://en.wikipedia.org/wiki/Precompiled_header){:target="_blank"}.
+They make compilation even more efficient than using #pragmas.
+
+The default Wiring template creates a few files (stdafx.h and stdafx.cpp) in your project that are used to manage precompiled headers.
+You can find more info about these files [here](http://msdn.microsoft.com/en-us/library/h552b3ca.aspx){:target="_blank"}.
```
Comment (ooeygui) [inline VisualStudio.md:24]: We've explicitly disabled them, so you should mention that. We can't enable them due to Arduino headers not including them

File: ShieldCompat.md (hunk 1 above)
Comment (IoTDan) [inline ShieldCompat.md]: updated.
[&laquo; Return to Homepage](index.htm){: .btn .btn-default}

File: VisualStudio.md (hunk 2 above)
Comment (IoTDan) [inline VisualStudio.md]: updated! 
[&laquo; Return to Homepage](index.htm){: .btn .btn-default}

File: VisualStudio.md (hunk 3 above)
Comment (IoTDan) [inline VisualStudio.md:24]: Done!!!!

**Pull request merged**